lib/hanami/model/config/adapter.rb

```diff
@@ -103,13 +103,10 @@ def instantiate_adapter(mapper)
           begin
             klass = Hanami::Utils::Class.load!(class_name, Hanami::Model::Adapters)
             klass.new(mapper, uri, options)
-          rescue NameError
-            raise AdapterNotFound.new(class_name)
```

Comment:
Got rid of this because:
1. It hides NoMethodErrors, which are quite common because of `nil`...
2. The obvious purpose, to catch undefined classes, has a check in `load_adapter`, which is called right before this. They're both private methods, so the class should always be defined by the time the code gets here.